x-pack/plugins/alerting/server/rules_client/types.ts

```diff
@@ -80,18 +83,29 @@ export interface RulesClientContext {
   readonly connectorAdapterRegistry: ConnectorAdapterRegistry;
   readonly getAlertIndicesAlias: GetAlertIndicesAlias;
   readonly alertsService: AlertsService | null;
+  readonly isSystemAction: (actionId: string) => boolean;
 }
 
-export type NormalizedAlertAction = Omit<RuleAction, 'actionTypeId'>;
+export type NormalizedAlertAction = DistributiveOmit<RuleAction, 'actionTypeId'>;
+export type NormalizedSystemAction = Omit<RuleSystemAction, 'actionTypeId'>;
```

Comment:
@cnasikas Why in one case you use `DistributiveOmit` and in another `Omit`?

Reply:
The reason is that the `RuleAction` is a union`RuleDefaultAction | RuleSystemAction`. Normal `Omit` does not work with unions. If you need to remove a property for each item in the union you need to use `DistributiveOmit`. For `RuleSystemAction` is not needed because is not a union.